sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/XmlSensorTest.java

```diff
@@ -76,6 +77,16 @@
 
   private final RuleKey parsingErrorCheckRuleKey = RuleKey.of(CheckRepository.REPOSITORY_KEY, parsingErrorCheckKey);
 
+  @Test
+  public void testPerformance() throws Exception {
+    createXmlFile(20000, "smallFile.xml");
+    long timeSmallFile = measureTimeToAnalyzeFile();
```

Comment:
OK so it's not going to work as you expect... Or I don't understand what you are trying to test.

The first time you call `measureTimeToAnalyzeFile`, there is only `smallFile.xml` in fileSystem, so the measure is correct. However, the second time you call it, there is now `smallFile.xml` **AND** `bugFile.xml`, so tests are not isolated and you are actually reanalyzing AGAIN `smallFile.xml`.

Make sure to have only a single file in FileSystem before measuring each file (or prove me wrong).

Reply:
Okay, perfect! then it is only a design problem. From my point of view the methods are doing way too many things which can not be extrapolated from their names.

Could you separate initialization of FileSystem and creation of XML file? so it's explicit in the test? 
I would go for something such as : `initFileSystemWithFile(createXmlFile(...))`

Reply:
Fixed according to your comment.